### ISSUE TITLE
Sonobuoy is still running, Got a duplicate result e2e

### ISSUE THREAD

**Issue by stamak** [heptio/sonobuoy#399]
**BUG REPORT**:

**What happened**:

```
sonobuoy status
PLUGIN          STATUS          COUNT
e2e             complete        1
e2e             running         1
systemd_logs    complete        4
systemd_logs    running         4

Sonobuoy is still running. Runs can take up to 60 minutes.
```

Logs:
```
sonobuoy logs -f
namespace="heptio-sonobuoy" pod="sonobuoy" container="kube-sonobuoy"
time="2018-03-27T13:50:42Z" level=info msg="Scanning plugins in ./plugins.d (pwd: /)"
time="2018-03-27T13:50:42Z" level=info msg="Scanning plugins in /etc/sonobuoy/plugins.d (pwd: /)"
time="2018-03-27T13:50:42Z" level=info msg="Directory (/etc/sonobuoy/plugins.d) does not exist"
time="2018-03-27T13:50:42Z" level=info msg="Scanning plugins in ~/sonobuoy/plugins.d (pwd: /)"
time="2018-03-27T13:50:42Z" level=info msg="Directory (~/sonobuoy/plugins.d) does not exist"
time="2018-03-27T13:50:42Z" level=info msg="Scanning plugins in ./plugins.d (pwd: /)"
time="2018-03-27T13:50:42Z" level=info msg="Scanning plugins in /etc/sonobuoy/plugins.d (pwd: /)"
time="2018-03-27T13:50:42Z" level=info msg="Directory (/etc/sonobuoy/plugins.d) does not exist"
time="2018-03-27T13:50:42Z" level=info msg="Scanning plugins in ~/sonobuoy/plugins.d (pwd: /)"
time="2018-03-27T13:50:42Z" level=info msg="Directory (~/sonobuoy/plugins.d) does not exist"
time="2018-03-27T13:50:42Z" level=info msg="Filtering namespaces based on the following regex:.*|heptio-sonobuoy"
time="2018-03-27T13:50:42Z" level=info msg="Namespace default Matched=true"
time="2018-03-27T13:50:42Z" level=info msg="Namespace e2e-tests-pod-network-test-k4cwr Matched=true"
time="2018-03-27T13:50:42Z" level=info msg="Namespace e2e-tests-projected-9nzzh Matched=true"
time="2018-03-27T13:50:42Z" level=info msg="Namespace e2e-tests-sched-pred-5k9j6 Matched=true"
time="2018-03-27T13:50:42Z" level=info msg="Namespace heptio-sonobuoy Matched=true"
time="2018-03-27T13:50:42Z" level=info msg="Namespace kube-public Matched=true"
time="2018-03-27T13:50:42Z" level=info msg="Namespace kube-system Matched=true"
time="2018-03-27T13:50:42Z" level=info msg="Starting server Expected Results: [{ e2e} {k8s-node0-syndeo-k8se2e-31-sjc1 systemd_logs} {
k8s-node1-syndeo-k8se2e-31-sjc1 systemd_logs} {k8s-node2-syndeo-k8se2e-31-sjc1 systemd_logs} {k8s-node3-syndeo-k8se2e-31-sjc1 systemd_
logs} { e2e} {k8s-node0-syndeo-k8se2e-31-sjc1 systemd_logs} {k8s-node1-syndeo-k8se2e-31-sjc1 systemd_logs} {k8s-node2-syndeo-k8se2e-31
-sjc1 systemd_logs} {k8s-node3-syndeo-k8se2e-31-sjc1 systemd_logs}]"
time="2018-03-27T13:50:42Z" level=info msg="Running plugin" plugin=e2e
time="2018-03-27T13:50:42Z" level=info msg="starting aggregation server" address=0.0.0.0 port=8080
time="2018-03-27T13:50:46Z" level=info msg="Running plugin" plugin=systemd-logs
time="2018-03-27T13:50:47Z" level=info msg="Running plugin" plugin=e2e
time="2018-03-27T13:50:47Z" level=info msg="Running plugin" plugin=systemd-logs
......

ime="2018-03-27T14:56:57Z" level=info msg="Running ns query (e2e-tests-sched-pred-5k9j6)"
time="2018-03-27T14:56:59Z" level=info msg="Collecting Pod Logs..."
time="2018-03-27T14:56:59Z" level=info msg="Collecting Pod Logs..."
time="2018-03-27T14:56:59Z" level=error msg="error querying PodPresets: the server could not find the requested resource (get podprese
ts.settings.k8s.io)"
time="2018-03-27T14:56:59Z" level=error msg="error querying PodPresets: the server could not find the requested resource (get podprese
ts.settings.k8s.io)"
time="2018-03-27T14:57:02Z" level=info msg="Running ns query (heptio-sonobuoy)"
time="2018-03-27T14:57:03Z" level=info msg="Collecting Pod Logs..."
time="2018-03-27T14:57:08Z" level=error msg="error querying PodPresets: the server could not find the requested resource (get podprese
ts.settings.k8s.io)"
time="2018-03-27T14:57:10Z" level=info msg="Running ns query (kube-public)"
time="2018-03-27T14:57:12Z" level=info msg="Collecting Pod Logs..."
time="2018-03-27T14:57:12Z" level=info msg="Collecting Pod Logs..."
time="2018-03-27T14:57:12Z" level=error msg="error querying PodPresets: the server could not find the requested resource (get podprese
ts.settings.k8s.io)"
time="2018-03-27T14:57:12Z" level=error msg="error querying PodPresets: the server could not find the requested resource (get podprese
ts.settings.k8s.io)"
time="2018-03-27T14:57:15Z" level=info msg="Running ns query (kube-system)"
time="2018-03-27T14:57:18Z" level=info msg="Collecting Pod Logs..."
time="2018-03-27T14:57:29Z" level=info msg="Collecting Pod Logs..."
time="2018-03-27T14:57:39Z" level=error msg="error querying PodPresets: the server could not find the requested resource (get podprese
ts.settings.k8s.io)"
time="2018-03-27T14:57:39Z" level=error msg="error querying PodPresets: the server could not find the requested resource (get podprese
ts.settings.k8s.io)"
time="2018-03-27T14:57:44Z" level=info msg="Results available at /tmp/sonobuoy/201803271350_sonobuoy_0c8484c0-e392-47f4-98f8-ff5c1d80f
6f4.tar.gz"
time="2018-03-27T14:57:44Z" level=info msg="no-exit was specified, sonobuoy is now blocking"
time="2018-03-27T14:57:49Z" level=info msg="received aggregator request" client_cert=e2e plugin_name=e2e
time="2018-03-27T14:57:49Z" level=warning msg="Got a duplicate result e2e"
```

**What you expected to happen**:

Tests are finished

**How to reproduce it (as minimally and precisely as possible)**:

`sonobuoy run` and wait

**Environment**:
```
sonobuoy version
v0.11.0
```
- Kubernetes version (use `kubectl version`): 
```
kubectl version
Client Version: version.Info{Major:"1", Minor:"9", GitVersion:"v1.9.1", GitCommit:"3a1c9449a956b6026f075fa3134ff92f7d55f812", GitTreeState:"clean", BuildDate:"2018-01-04T11:52:23Z", GoVersion:"go1.9.2", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"9", GitVersion:"v1.9.4+coreos.0", GitCommit:"5e73e0769e5c4ac497235e2817868b1a37032fba", GitTreeState:"clean", BuildDate:"2018-03-12T20:05:58Z", GoVersion:"go1.9.3", Compiler:"gc", Platform:"linux/amd64"}
````

- Cloud provider or hardware configuration: OpenStack
- OS (e.g. from /etc/os-release): Container Linux by CoreOS 1520.9.0
- Kernel (e.g. `uname -a`): `4.14.19-coreos`

**Comment by chuckha** [comment]
In this instance the duplicate result message is expected. We only accept the first plugin response. If we get a response from the same plugin we write out the "Got a duplicate result" message.

However, the strange thing here is that it looks like two copies of all the plugins and two copies of the aggregator server are running. How did you run sonobuoy? Would you happen to have the YAML available for us to look at?

**Comment by stamak** [comment]
@chuckha Yes , strange because I used rc2 and I had everything ok ( 1 copy) and now switch back rc2 I do not have this problem 

```
stamak$ ./sonobuoy version
v0.11.0-rc.2
stamak$ ./sonobuoy status
PLUGIN          STATUS          COUNT
e2e             running         1
systemd_logs    complete        4

Sonobuoy is still running. Runs can take up to 60 minutes.
```

I am using command `sonobuoy run`

**Comment by stamak** [comment]
v0.11.0

```
stamak$ sonobuoy --debug version
v0.11.0
stamak$ sonobuoy --debug run
Running plugins: e2e, systemd-logs
INFO[0007] created object                                name=heptio-sonobuoy namespace= resource=namespaces
INFO[0008] created object                                name=sonobuoy-serviceaccount namespace=heptio-sonobuoy resource=serviceaccounts
INFO[0008] created object                                name=sonobuoy-serviceaccount-heptio-sonobuoy namespace= resource=clusterrolebindings
INFO[0008] created object                                name=sonobuoy-serviceaccount namespace= resource=clusterroles
INFO[0008] created object                                name=sonobuoy-config-cm namespace=heptio-sonobuoy resource=configmaps
INFO[0009] created object                                name=sonobuoy-plugins-cm namespace=heptio-sonobuoy resource=configmaps
INFO[0009] created object                                name=sonobuoy namespace=heptio-sonobuoy resource=pods
INFO[0009] created object                                name=sonobuoy-master namespace=heptio-sonobuoy resource=services
stamak$ sonobuoy --debug status
PLUGIN          STATUS          COUNT
e2e             running         2
systemd_logs    complete        4
systemd_logs    running         4

Sonobuoy is still running. Runs can take up to 60 minutes.
```

@chuckha Can you help me how I can get yaml to you?

**Comment by mwieczorek** [comment]
I got the same issue. Cluster on Azure, made with [acs-engine](https://github.com/Azure/acs-engine)

**Comment by chuckha** [comment]
The issue here is how we are loading plugins.

We do a search across many paths and assume any *.yml or *.yaml file we find in there is a plugin to load. Unfortunately some code is duplicating the SearchPaths which means it finds all plugins twice and therefore loads all plugins twice.

I'm working on a fix right now, thanks for reporting this issue.

**Comment by chuckha** [comment]
Fixed and released with version v0.11.1
